web/models/requirement.ex

```diff
@@ -4,81 +4,79 @@ defmodule HexWeb.Requirement do
   schema "requirements" do
     field :app, :string
     field :requirement, :string
-    field :optional, :boolean
+    field :optional, :boolean, default: false
 
-    # The name of the dependency
+    # The name of the dependency used to find the package
     field :name, :string, virtual: true
 
     belongs_to :release, Release
     belongs_to :dependency, Package
   end
 
-  # TODO: Clean this up after with lands in ecto
+  def changeset(requirement, params \\ %{}) do
+    changeset = cast(requirement, params, ~w(name app requirement optional))
 
-  def create_all(release, requirements) do
-    requirements = normalize(requirements)
-    deps = deps(requirements)
+    name = changeset.changes.name
+    app = Map.get(changeset.changes, :app, name)
+    dep = HexWeb.Repo.get_by(Package, name: name)
```

Comment:
since `Requirement.changeset` is called from `Requirement.create_all` in a "loop" we'd end up with multiple queries whereas one query could do. Is there a way to optimize it? I kind of want to keep the `.changeset` function here because using `cast_assoc(:requirements)` is much easier than filling in everything manually.